app/src/shared/lib/hooks/useInfiniteScroll.ts

```diff
@@ -0,0 +1,55 @@
+import { RefObject, useCallback, useEffect, useRef } from 'react';
+
+/**
+ *
+ * @param rootElement - ссылка на корневой элемент, который необходим для наблюдения за пересечением с последним элементом
+ * @param trigger - триггер, при котором будет происходить перерасчет последнего элемента
+ * @param callback - функция, которая будет вызываться при пересенчении корневого элемента и последнего элемента
+ */
+
+export const useInfiniteScroll = <T extends HTMLElement | null>(
+    rootElement: RefObject<T>,
+    trigger: number | string | boolean,
+    callback: () => void,
+) => {
+    const lastElement = useRef<ChildNode | null | undefined>(null);
```

Comment:
```suggestion
    const lastElement = useRef<Element| null | undefined>(null);
```
попробуй так
беспокоит `as Element`

Reply:
ts на тип ругается всё равно 
![image](https://github.com/user-attachments/assets/e96fb810-f71d-4b43-8010-b2d17a22744c)

Reply:
а если 'as Element' убрать, то ошибка типов на функции observer 
![image](https://github.com/user-attachments/assets/b4c6cc71-715f-4be7-a317-3598c9e23e55)